kafkareporter/kafka.go

```diff
@@ -297,6 +297,17 @@ func (r *kafkaReporter) Close() {
 		r.logger.Print(err)
 	}
 }
+//循环处理上报信息的结果信息，以实现kafka单次连接多次上报。
+func (r *kafkaReporter) readBackMessage() {
+	for {
+		select {
+		case <-r.producer.Successes():
+		case err := <- r.producer.Errors():
+			r.logger.Fatalf( "kafkaReport readBackMessage error detail is %s",err)
```

Comment:
I think using the `log.Errorf` output will do, which will cause the process to end

Reply:
You can check the revised document to confirm.    github.com/Shopify/sarama/async_producer.go

	// Successes is the success output channel back to the user when Return.Successes is
	// enabled. If Return.Successes is true, you MUST read from this channel or the
	// Producer will deadlock. It is suggested that you send and read messages
	// together in a single select statement.
	Successes() <-chan *ProducerMessage

	// Errors is the error output channel back to the user. You MUST read from this
	// channel or the Producer will deadlock when the channel is full. Alternatively,
	// you can set Producer.Return.Errors in your config to false, which prevents
	// errors to be returned.
	Errors() <-chan *ProducerError

Reply:
https://github.com/golang/go/blob/go1.16.15/src/log/log.go#L209-L212

@liufugen-yewu The process will exit after `Fatalf`